mlir/lib/Dialect/MIOpen/Transforms/GridwiseGemmToBlockwise.cpp

```diff
@@ -2496,18 +2496,22 @@ struct GridwiseGemmV2RewritePattern
 
       llvm::copy(tailResults, std::back_inserter(transformedTail));
     }
-    // Slice up vectors here to make it clearer that each store loop
-    // deals with a distinct set of values.
-    llvm::SmallVector<Value, 4> vectors;
-    vectors.reserve(transformedTail.size() * iterationsPerVectorC);
+
+    // Merge the vectors using miopen.insertslice and store loop to use
+    // miopen.extractslice so offset parsing can be deferred after loop
+    // unrolling.
+    int64_t vectorLen = vectorType.getShape()[0];
+    VectorType mergedType =
+        VectorType::get({static_cast<long>(vectorLen * transformedTail.size())},
+                        vectorType.getElementType());
+    Value resultMerged =
```

Comment:
I'd like a perf test here - I'm hoping this early post-swizzle manipulation doesn't screw up the instruction selector

Reply:
Yeah, I'm also interested if this causes any performance regression. Fortunately, few cases I've seen showed almost identical instructions before going into llvm already. Let's see the performance report and I'll investigate if it catches anything.

Reply:
I'd give some convolutions that use swizzles a `--debug-only=serialize-to-blob` from this PR and from `miopen-dialect`

Also, for perf, you can run the perf test script yourself: you need to `ninja ci-performance-scripts` and then do something like
```
./bin/MIOpenDriver.py -o nhwc-test-description.csv -F 1 -f NHWC -n 256 -c 128 -H 28 -W 28 -k 128 -x 3 -y 3 -p 1 -q 1 -u 1 -v 1 -l 1 -j 1 -g 1
```

(that exact set of parameters is why I ended up not doing the swizzling on one big vector/all at once)

Reply:
Performance is not changing.
I've noticed that there are few convolutions sometimes run very slowly. including (x2~x4 slower than usual)
with or without this PR. haven't got chance to investigate it yet but includes these

bwd,f16,True,kcyx,nchw,nkhw,256,256,14,14,256,3,3,1,1,1,1,1,1,4.35545105968997
wrw,f32,True,kcyx,nchw,nkhw,256,512,16,16,512,3,3,1,1,2,2,0,0,5.58864260967153,

Excepting that average performance is 0.2% better but I interpret it's just the same.